builtin/providers/azurerm/resource_arm_servicebus_queue.go

```diff
@@ -0,0 +1,229 @@
+	client := meta.(*ArmClient).serviceBusQueuesClient
+	log.Printf("[INFO] preparing arguments for Azure ARM ServiceBus Queue creation.")
+
+	name := d.Get("name").(string)
+	namespaceName := d.Get("namespace_name").(string)
+	location := d.Get("location").(string)
+	resGroup := d.Get("resource_group_name").(string)
+
+	parameters := servicebus.QueueCreateOrUpdateParameters{
+		Name:            &name,
+		Location:        &location,
+		QueueProperties: &servicebus.QueueProperties{},
+	}
+
+	if autoDeleteOnIdle := d.Get("auto_delete_on_idle").(string); autoDeleteOnIdle != "" {
+		parameters.QueueProperties.AutoDeleteOnIdle = &autoDeleteOnIdle
+	}
+
+	if defaultTTL := d.Get("default_message_ttl").(string); defaultTTL != "" {
+		parameters.QueueProperties.DefaultMessageTimeToLive = &defaultTTL
+	}
+
+	if duplicateWindow := d.Get("duplicate_detection_history_time_window").(string); duplicateWindow != "" {
+		parameters.QueueProperties.DuplicateDetectionHistoryTimeWindow = &duplicateWindow
+	}
+
+	enableBatchedOps := d.Get("enable_batched_operations").(bool)
+	enableExpress := d.Get("enable_express").(bool)
+	enablePartitioning := d.Get("enable_partitioning").(bool)
+	maxSize := int64(d.Get("max_size_in_megabytes").(int))
+	requiresDuplicateDetection := d.Get("requires_duplicate_detection").(bool)
+	supportOrdering := d.Get("support_ordering").(bool)
+
+	parameters.QueueProperties.EnableBatchedOperations = &enableBatchedOps
+	parameters.QueueProperties.EnableExpress = &enableExpress
+	parameters.QueueProperties.EnablePartitioning = &enablePartitioning
+	parameters.QueueProperties.MaxSizeInMegabytes = &maxSize
+	parameters.QueueProperties.RequiresDuplicateDetection = &requiresDuplicateDetection
+	parameters.QueueProperties.SupportOrdering = &supportOrdering
+
+	_, err := client.CreateOrUpdate(resGroup, namespaceName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.Get(resGroup, namespaceName, name)
+	if err != nil {
+		return err
+	}
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read ServiceBus Queue %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmServiceBusQueueRead(d, meta)
+}
+
+func resourceArmServiceBusQueueRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusQueuesClient
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	name := id.Path["queues"]
+
+	resp, err := client.Get(resGroup, namespaceName, name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure ServiceBus Queue %s: %s", name, err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("namespace_name", namespaceName)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+
+	props := resp.QueueProperties
```

Comment:
I think we should check that `QueueProperties` is not nil before trying to get values from it - otherwise this could panic